src/transpiler.ts

```diff
@@ -1,9 +1,28 @@
 import Generator from './generator/generator';
 import Parser from './parser/parser';
 
+// noinspection TsLint
+var performance: any;
+
+// Only require performance when running under nodejs
+if (typeof global === 'object') {
+    // noinspection TsLint
+    performance = require('perf_hooks').performance;
+} else {
+    performance = window.performance;
+}
+
+interface Timing {
+    compilationTime: number;
+    importTime: number;
+    executionTime: number;
+}
+
 class Transpiler {
```

Comment:
I think it would be better to not use static methodes any longer to encapsulate the time values in an object. Otherwise if we have to expand this code it will get much more time consuming to avoid inconsistencies with the static members.

Reply:
So you suggest changing both `transpile` and `getTiming` to non-static methods?

Reply:
Exactly.

Reply:
The problem is that `transpile` does not use any state and should therefore be non-static.

Reply:
If we change it to a non-static version we could use a subclass for the time measurements. Therefore the production code doesn't get bloated with measurement code. The drawback is that one has to create a new `Transpiler` instance and not just call a static member function. But in my opinion this is the lesser evil